### PR TITLE
Login form & login end point - Work in Progess

### PR DESCRIPTION
**Pull request by Leidagandy**
Thank you for contributing! To help the review process, please provide the following:

### Proposal
Provide a short description of the change you are proposing and the reasons for it.

### Related
Provide links to any associated pull requests or issues. You can use `#123` to link to a PR or issue in this repository, or `user/repo#123` to link to other repositories.

### Checklist

- [ ] I have read the [code of conduct] and [contributing guide]
- [ ] I have made this pull request to the `master` branch
- [ ] I have run all of the automated validation using `npm run ship`
- [ ] I have added myself to the `"contributors"` list in the `package.json` (or do not want to)

[code of conduct]: https://github.com/textbook/starter-kit/blob/master/.github/CODE_OF_CONDUCT.md
[contributing guide]: https://github.com/textbook/starter-kit/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

File: client/src/components/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  const [message, setMessage] = useState("");
+  const [validUser, setValidUser] = useState(false);
+
+  const history = useHistory();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    axios
+      .post("api/login", {
```
Comment (Dedekind561) [inline client/src/components/LoginForm.js:18]:
```suggestion
      .post("/api/login", {
```

File: client/src/components/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import axios from "axios";
+import "./LoginForm.css";
+
+function LoginForm() {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
+  const [validUser, setValidUser] = useState(false);
+
+  const history = useHistory();
```
Comment (Dedekind561) [inline client/src/components/LoginForm.js:12]: What is this `history` for ?

File: client/src/components/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+      })
+      .catch(function (error) {
+        if (error) {
+          setEmail("");
+          setPassword("");
+          setMessage("Invalid email or password!");
+        }
+        console.log(error);
+      });
+    // console.log("data from: ", email, password)
```
Comment (Dedekind561) [inline client/src/components/LoginForm.js:37]: Make sure you remove any rogue comments

File: server/api.js [hunk 4]
```diff
@@ -1,82 +1,109 @@
...
-
 router.get("/", (_, res, next) => {
-
-	Connection.connect((err) => {
-		if (err) {
-			return next(err);
-		}
-		res.json({ message: "Hello, world!" });
-	});
+  Connection.connect((err) => {
```
Comment (Dedekind561) [inline server/api.js:5]:
```suggestion
 connection.connect((err) => {
```
Make this `connection` variable with a lower case letter at the start

File: server/api.js [hunk 5]
```diff
@@ -1,82 +1,109 @@
...
+    res.json({ message: "Hello, Compass team!" });
+  });
 });
 
+router.post("/login", (req, res, next) => {
+  const email = req.body.email;
+  const password = req.body.password;
+
+  if (email && password) {
+    Connection.query(
```
Comment (Dedekind561) [inline server/api.js:18]:
```suggestion
    connection.query(
```

File: server/api.js [hunk 6]
```diff
@@ -1,82 +1,109 @@
...
+      "select * from students where email = $1 and password = $2",
+      [email, password],
+      (err, result) => {
+        if (result.rowCount > 0) {
+          return res.status(200).send(result.rows[0]);
+        } else {
```
Comment (Dedekind561) [inline server/api.js:24]: What is this `else` block for ?

File: server/api.js [hunk 7]
```diff
@@ -1,82 +1,109 @@
...
 router.get("/students/:id", (_, res, next) => {
-	let studentId = Number(_.params.id);
-	console.log("Hi");
-	Connection.query("SELECT * FROM students WHERE id = $1", [studentId],(err,result) => {
-		if (err) {
-			return next(err);
-		}
-		res.json(result.rows[0]);
-	});
+  let studentId = Number(_.params.id);
```
Comment (Dedekind561) [inline server/api.js:46]: Good variable naming here 👍

File: server/api.js [hunk 8]
```diff
@@ -1,82 +1,109 @@
...
+  console.log("Hi");
+  Connection.query(
```
Comment (Dedekind561) [inline server/api.js:48]:
```suggestion
  connection.query(
```

File: server/api.js [hunk 9]
```diff
@@ -1,82 +1,109 @@
...
-	let studentId = Number(_.params.id);
-	console.log("Hi");
-	Connection.query("SELECT * FROM students WHERE id = $1", [studentId],(err,result) => {
-		if (err) {
-			return next(err);
-		}
-		res.json(result.rows[0]);
-	});
+  let studentId = Number(_.params.id);
+  console.log("Hi");
```
Comment (Dedekind561) [inline server/api.js:47]: Remove old `console.logs` that you don't need anymore

**Review by Dedekind561 — CHANGES_REQUESTED** [review]
Great start, folks - a few minor changes here suggested in here :)

File: client/src/components/LoginForm.js (hunk 2 above)
Comment (SULAIMAN-100) [reply client/src/components/LoginForm.js:12]: this is a react-router independency, we used it here to switch the paths

File: server/api.js (hunk 6 above)
Comment (SULAIMAN-100) [reply server/api.js:24]: first checking the students' table if not exists check-in mentors' table if not exist as well return an error.

**Pull request merged**